script/airdrop-monitor.sh

```diff
@@ -0,0 +1,63 @@
+# This script monitors the Airdrop canister for new parcipants list,
+# and sends ethereum transaction to distribute ICP ERC20 tokens.
+#
+# Environment variables must be set:
+#
+# ETH_RPC_URL   RPC provider's URL
+#
+# ETH_FROM      The address to sign & send airdrop transactions.
+#               It is also the owner of Airdrop contract.
+#               Lastly seth must import the private key first.
+#
+# AIRDROP_FROM  The address of the Airdrop contract
+
+AIRDROP_CLAIMED_JSON="airdrop_claimed.json"
+AIRDROP_CLAIMED_HASH="airdrop_claimed.hash"
+INTERVAL_SECS=60
+
+test -z "$ETH_RPC_URL" && echo "Must set ETH_RPC_URL" && exit 1
+test -z "$ETH_FROM" && echo "Must set ETH_FROM" && exit 1
+test -z "$AIRDROP_FROM" && echo "Must set AIRDROP_FROM" && exit 1
+
+function sync_participant_list() {
+    echo "[[\"0x2c91e73a358e6f0aff4b9200c8bad0d4739a70dd\", \"100000000000000000\"], [\"0x2c91e73a358e6f0aff4b9200c8bad0d4739a70dd\", \"100000000000000000\"]]"
+}
+
+while true; do
+    echo 1. Syncing participant list...
+    LIST=$(sync_participant_list)
+    if [[ -z "$LIST" ]]; then
+        echo '   No new participants'
+    else
+        NUM=$(echo "$LIST" | jq length)
+        ADDRS=$(echo "$LIST" | jq 'map(.[0])' | sed -e 's/"//g')
+        AMOUNTS=$(echo "$LIST" | jq 'map(.[1])' | sed -e 's/"//g')
+        echo "  Synced, and found $NUM new participants."
+
+        while true; do
+            echo 2. Creating the airdrop transaction...
+            NONCE=$(seth nonce "$ETH_FROM")
+            TX=$(seth --nonce "$NONCE" -S/dev/null mktx "$AIRDROP_FROM" "airdrop(address[],uint256[])" "${ADDRS}" "${AMOUNTS}")
```

Comment:
It seems that `--prio-fee` is required (and can be set to 2 gwei, should work in most cases).
https://github.com/dapphub/dapptools/blob/master/src/seth/README.md#seth-send

Reply:
Is this a mainnet requirement? Seems not required on Goerli.

Reply:
I have not tested it myself, just commenting based on docs.